### ISSUE TITLE
XCode 16 - Undefined symbols in UnityFramework

### ISSUE THREAD

**Issue by BernoAU** [yasirkula/UnityNativeGallery#337]
**Description of the bug**

After upgrading to XCode 16 my project has undefined symbols for functions used in UnityNativeGallery.

**Reproduction steps**

Update to XCode 16 and build a Unity project using UnityNativeGallery.
Open the created XCode project and the following errors are shown:
![0293FF0C-E645-412C-AC6D-22CE4E6439BB_1_201_a](https://github.com/user-attachments/assets/a587a4a2-a359-4f07-8448-85ffda9923c5)

**Platform specs**

- Unity version: 2020.3.48f1
- Platform: *iOS*
- Device: none
- How did you download the plugin: GitHub

**Comment by BernoAU** [comment]
I have solved this by modifying the file: NativeGallery\Editor\NGPostProcessBuild.cs

```
pbxProject.RemoveFrameworkFromProject( targetGUID, "Photos.framework" );
pbxProject.RemoveFrameworkFromProject( targetGUID, "PhotosUI.framework" );

//EDIT: XCode 16 requires this framework reference
pbxProject.AddFrameworkToProject(targetGUID, "ImageIO.framework", true);

File.WriteAllText( pbxProjectPath, pbxProject.WriteToString() );
```

**Comment by yasirkula** [comment]
This line should've done the same thing: https://github.com/yasirkula/UnityNativeGallery/blob/4d27475bed66d56aca8b7d585b98fe503525e593/Plugins/NativeGallery/Editor/NGPostProcessBuild.cs#L128

Can you reproduce this issue on new Unity projects?

**Comment by BernoAU** [comment]
No. I just built a bare bones project with Unity 6 and it did not have this error.
Maybe a Unity 2020 issue?

**Comment by yasirkula** [comment]
Sorry, by new project I meant fresh/clean Unity project on the same Unity version. It'll very likely work on 2020 as well. My best best is that in your project, another plugin is calling SetBuildProperty to override `OTHER_LDFLAGS` and it is extremely irritating and irresponsible of that plugin's developer. They are not the only person to have access to the Xcode project and have no right to override other plugins' options. I'd recommend googling SetBuildProperty in your project.

**Comment by BernoAU** [comment]
I use AddBuildProperty myself in a postbuild script but not to modify OTHER_LDFLAGS.
The only other script I could find modifying OTHER_LDFLAGS is the postbuild for TextureOps.
According to the doco this shouldn't be the problem as duplicate build properties are ignored:
https://docs.unity3d.com/2020.3/Documentation/ScriptReference/iOS.Xcode.PBXProject.AddBuildProperty.html

**Comment by yasirkula** [comment]
Hmm, then I believe the culprit is located inside a dll. Right now, your solution is the way to go. But you should call AddFrameworkToProject for all of these frameworks: https://github.com/yasirkula/UnityNativeGallery/blob/4d27475bed66d56aca8b7d585b98fe503525e593/Plugins/NativeGallery/Editor/NGPostProcessBuild.cs#L116-L120

The ones using `-weak_framework` should have **weak** parameter set to true.

**Comment by BernoAU** [comment]
OK I have found the cause.
It was starting to drive me crazy as sometimes the frameworks were added and other times they were not.
I use an editor script that calls `EditorUserBuildSettings.SwitchActiveBuildTarget` to automate the process of building for multiple platforms. This function is not working correctly anymore with Unity 2020.3.48f1. 
Conditional code for each platform isn't being activated when it switches platform.
I thought my editor script fixed it but it was just because I manually built for iOS after making the change!!
It's Unity's bug not yours, sorry for the confusion.

**Comment by yasirkula** [comment]
I'm glad to hear it! As far as I know, this bug exists on all Unity versions, perhaps even on Unity Cloud Build. In your editor script, I'd recommend actually changing the platform and waiting for the code to recompile. To start the build process after recompilation, you can use a combination of SessionState and InitializeOnLoadMethod.

**Comment by BernoAU** [comment]
Thanks so much for the tip about SessionState I didn't know about that. Thanks to that I could automate my builds and all editor preprocessing is performed correctly!
Thanks for sharing your code as well. I am using both Native Gallery and Texture ops its fantastic work.
